addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/internal/protocol/message/MySensorsMessage.java

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) 2014-2016 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mysensors.internal.protocol.message;
+
+import static org.openhab.binding.mysensors.MySensorsBindingConstants.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Used to store the content of a MySensors message.
+ *
+ * @author Tim Oberföll
+ *
+ */
+public class MySensorsMessage {
+    public static String GATEWAY_STARTUP_NOTIFICATION = "Gateway startup complete.";
+
+    private Logger logger = LoggerFactory.getLogger(MySensorsMessage.class);
+
+    public int nodeId = 0; // id of the node in the MySensors network
+    public int childId = 0; // id of the child of the node (more than one possible)
+    public int msgType = 0; // type of message: request, internal, presentation ...
+    public int ack = 0; // is an acknoledgement requested?
+    public boolean revert = true; // revert status if no ack was received from the node
+    public int subType = 0; // like: humidity, temperature, light ...
+    public String msg = ""; // content of the message
+    public String oldMsg = ""; // content of the last message send to or received by the node
+    public int retries = 0; // number of retries if a message is not acknowledged by the receiver
+    public long nextSend = 0; // timestamp when the message should be send
+
+    public MySensorsMessage() {
+
+    }
+
+    public MySensorsMessage(int nodeId, int childId, int msgType, int ack, boolean revert, int subType, String msg) {
+        this.nodeId = nodeId;
+        this.childId = childId;
+        this.msgType = msgType;
+        this.ack = ack;
+        this.revert = revert;
+        this.subType = subType;
+        this.msg = msg;
+    }
+
+    /**
+     * Write message to DEBUG.
+     */
+    public void printDebug() {
+        logger.debug(String.format("nodeId: %d, childId: %d, msgType: %d, ack: %d, revert: %b, subType: %d ,msg: %s",
```

Comment:
could this be replaced by logger.debug(getDebugInfo())? the two strings look very similiar.